internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -286,72 +296,13 @@ func resourceRecoveryServicesVaultCreate(d *pluginsdk.ResourceData, meta interfa
 		return fmt.Errorf("creating %s: %+v", id.String(), err)
 	}
 
-	// `encryption` needs to be set before `cross_region_restore_enabled` is set. Or the service will return an error. "If CRR is enabled for the Vault, the storage state will be locked and it will interfere with further operations"
-	// recovery vault's encryption config cannot be set while creation, so a standalone update is required.
-	if _, ok := d.GetOk("encryption"); ok {
-		encryption, err := expandEncryption(d)
-		if err != nil {
-			return err
-		}
-		requireAdditionalUpdate = true
-		updatePatch.Properties.Encryption = encryption
-	}
-
 	if requireAdditionalUpdate {
 		err := client.UpdateThenPoll(ctx, id, updatePatch)
 		if err != nil {
 			return fmt.Errorf("updating Recovery Service %s: %+v, but recovery vault was created, a manually import might be required", id.String(), err)
 		}
 
 	}
-
-	storageType := backupresourcestorageconfigsnoncrr.StorageType(d.Get("storage_mode_type").(string))
-	storageCfg := backupresourcestorageconfigsnoncrr.BackupResourceConfigResource{
-		Properties: &backupresourcestorageconfigsnoncrr.BackupResourceConfig{
-			StorageModelType:       &storageType,
-			CrossRegionRestoreFlag: utils.Bool(d.Get("cross_region_restore_enabled").(bool)),
-		},
-	}
-
-	err = pluginsdk.Retry(d.Timeout(pluginsdk.TimeoutCreate), func() *pluginsdk.RetryError {
-		if resp, err := storageCfgsClient.Update(ctx, storageId, storageCfg); err != nil {
-			if response.WasNotFound(resp.HttpResponse) {
-				return pluginsdk.RetryableError(fmt.Errorf("updating Recovery Service Storage Cfg %s: %+v", id.String(), err))
-			}
-
-			return pluginsdk.NonRetryableError(err)
-		}
-		return nil
-	})
-	if err != nil {
-		return fmt.Errorf("creating %s: %+v", id, err)
-	}
-
-	// storage type is not updated instantaneously, so we wait until storage type is correct
-	err = pluginsdk.Retry(d.Timeout(pluginsdk.TimeoutCreate), func() *pluginsdk.RetryError {
```

Comment:
same here, this seems like a lot of retry related logic that's being removed, what's the logic here?

Reply:
samilar to the `encryption`, the storage settings required a seprate client to set, but it can and must be set via the Vault API for now. It's tested by `TestAccRecoveryServicesVault_ToggleCrossRegionRestore` test case